greet.py

```diff
@@ -0,0 +1,10 @@
+def greet(name=None):
+    if name is None:
+        string = 'Hello, my friend'
+    else:
+        if name.isupper():
+            string = 'HELLO, ' + name
+        else:
+            string = 'Hello, ' + str(name)
```

Comment:
why do you use str here but not above? why do you need str at all?

Reply:
Can I not use str if it is not specified in the test cases?